addon/components/polaris-color-picker.js

```diff
@@ -63,97 +64,117 @@ export default Component.extend({
 
     const backgroundColor = `rgba(${red}, ${green}, ${blue}, ${alpha})`;
     return htmlSafe(`background-color: ${backgroundColor};`);
-  }).readOnly(),
+  }
 
   /**
    * @private
    */
-  draggerX: computed('color.saturation', 'pickerSize', function() {
+  @(computed('color.saturation', 'pickerSize').readOnly())
+  get draggerX() {
     const {
       color: { saturation },
       pickerSize,
     } = this.getProperties('color', 'pickerSize');
     return clamp(saturation * pickerSize, 0, pickerSize);
-  }).readOnly(),
+  }
 
   /**
    * @private
    */
-  draggerY: computed('color.brightness', 'pickerSize', function() {
+  @(computed('color.brightness', 'pickerSize').readOnly())
+  get draggerY() {
     const {
       color: { brightness },
       pickerSize,
     } = this.getProperties('color', 'pickerSize');
     return clamp(pickerSize - brightness * pickerSize, 0, pickerSize);
-  }).readOnly(),
+  }
+
+  @computed('contentId')
+  get element() {
+    return document.querySelector(`#${this.contentId}`);
+  }
 
   didRender() {
-    this._super(...arguments);
+    super.didRender(...arguments);
+
+    let element = this.element;
+
+    if (isNone(element)) {
+      return;
+    }
 
     // Grab the size of the picker for positioning the draggable markers.
-    const mainColorElement = this.element.querySelector(
+    const mainColorElement = element.querySelector(
       'div.Polaris-ColorPicker__MainColor'
     );
     if (isNone(mainColorElement)) {
       return;
     }
 
     this.set('pickerSize', mainColorElement.clientWidth);
-  },
+  }
+
+  @action
+  setContentId(element, [value]) {
```

Comment:
Reckon this, then `contentId` property etc. are worth putting into a mixin? We're already using them 4 times in this PR alone 🤔